crates/biome_js_analyze/src/assist/source/use_sorted_interface_members.rs

```diff
@@ -0,0 +1,200 @@
+        {
+            sortable_keys.push(ComparableToken::new(token_text));
+        }
+    }
+
+    // Check if the sortable keys are already sorted
+    for i in 1..sortable_keys.len() {
+        if comparator(&sortable_keys[i - 1], &sortable_keys[i]) == std::cmp::Ordering::Greater {
+            return false;
+        }
+    }
+
+    true
+}
+fn sort_interface_members(
+    list: &TsTypeMemberList,
+    comparator: impl Fn(&ComparableToken, &ComparableToken) -> std::cmp::Ordering,
+) -> TsTypeMemberList {
+    let mut sortable_members = Vec::new();
+    let mut non_sortable_members = Vec::new();
+
+    // Separate sortable from non-sortable members
+    for member in list.iter() {
+        if let Ok(Some(name)) = member.name() {
+            if let Some(token_text) = name.name() {
+                sortable_members.push((member, ComparableToken::new(token_text)));
+            } else {
+                // Name exists but is not sortable (computed/dynamic)
+                non_sortable_members.push(member);
+            }
+        } else {
+            // No name (call signatures, construct signatures, index signatures)
+            non_sortable_members.push(member);
+        }
+    }
+
+    // Sort the sortable members
+    sortable_members.sort_by(|(_, a), (_, b)| comparator(a, b));
+
+    // Combine: all sortable members first, then all non-sortable members
+    let mut new_members: Vec<AnyTsTypeMember> = sortable_members
+        .into_iter()
+        .map(|(member, _)| member)
+        .collect();
+
+    new_members.extend(non_sortable_members);
+
+    make::ts_type_member_list(new_members)
+}
+declare_source_rule! {
+    /// Sort interface members by key.
+    ///
+    /// Interface members are sorted according to their names.
+    ///
+    /// This rule sorts all sortable members (properties, methods, getters, setters)
+    /// and groups them at the beginning of the interface. Non-sortable members
+    /// (call signatures, construct signatures, and index signatures) are grouped
+    /// at the end of the interface.
+    ///
+    /// # Examples
+    ///
+    /// ## Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface MixedMembers {
+    ///   z: string;
+    ///   a: number;
+    ///   (): void;  // Call signature - non-sortable
+    ///   y: boolean;
+    ///   b: string;
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```ts
+    /// interface MixedMembers {
+    ///   a: number;
+    ///   b: string;
+    ///   y: boolean;
+    ///   z: string;
+    ///   (): void;  // Non-sortable members grouped at end
+    /// }
+    /// ```
+    ///
+    pub UseSortedInterfaceMembers {
+        version: "2.0.0",
+        name: "useSortedInterfaceMembers",
+        language: "ts",
+        recommended: false,
+        sources: &[RuleSource::EslintPerfectionist("sort-interfaces").inspired()],
+        fix_kind: FixKind::Safe,
+    }
+}
+impl Rule for UseSortedInterfaceMembers {
```

Comment:
Generally, the code of all rules is structure like this;
- the `declare_*` macro
- the `Rule` trait
- the implementation of the rule

Please restructure the code to follow the other rules